src/main/java/by/exadel/internship/service/impl/UserTimeSlotServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+    private static int INTERVIEW_TIME = 30;
+    private static final String SIMPLE_CLASS_NAME = UserTimeSlotService.class.getSimpleName();
+
+
+    @Override
+    public void saveUserTime(UserDTO userDTO) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to save free time list for user with uuid = {} in DB", userDTO.getId());
+        List<UserTimeSlotWithUserDTO> resultUSerTimeList = new ArrayList<>();
+        userDTO.getUserTimeSlots().forEach(timeForCallUser -> {
+            checkTime(timeForCallUser);
+            separateTime(timeForCallUser, userDTO.getInterviewTime(), resultUSerTimeList);
+        });
+        resultUSerTimeList.forEach(time -> {
+            time.setUser(userDTO);
+        });
+        userTimeSlotRepository.saveAll(mapper.mapToEntity(resultUSerTimeList));
+        log.info("Free time list was saved in DB, user uuid = {}", userDTO.getId());
+    }
+
+    @Override
+    public void deletedById(UUID timeId) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Try to remove time with uuid = {} from user free time", timeId);
+        userTimeSlotRepository.deleteById(timeId);
+        log.info("Time with uuid = {} was deleted", timeId);
+    }
+
+    @Override
+    public void restoreUserTime(UserTimeSlotWithUserIdDTO time) {
+        MDCLog.putClassNameInMDC(SIMPLE_CLASS_NAME);
+        log.info("Restoring time with uuid = {} for user with uuid ={}", time.getId(), time.getUserId());
+        UserDTO userDTO = userService.getById(time.getUserId());
+        UserTimeSlotWithUserDTO userTimeSlotWithUserDTO = new UserTimeSlotWithUserDTO();
+        userTimeSlotWithUserDTO.setUser(userDTO);
+        userTimeSlotWithUserDTO.setStartDate(time.getStartDate());
+        switch (userDTO.getInterviewTime()) {
+            case HALF_HOUR: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(30));
+                break;
+            }
+            case HOUR: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(60));
+                break;
+            }
+            case HOUR_HALF: {
+                userTimeSlotWithUserDTO.setEndDate(time.getStartDate().plusMinutes(90));
+                break;
+            }
+        }
+        userTimeSlotRepository.save(mapper.toTimeForCallUserEntity(userTimeSlotWithUserDTO));
+        log.info("User time was restoring");
+    }
+
+    private void checkTime(UserTimeSlotDTO time) {
+        log.info("Bringing time to a common form");
+        time.setStartDate(commonTimeForm(time.getStartDate()));
+        time.setEndDate(commonTimeForm(time.getEndDate()));
+        log.info("Time brought to a common form");
+    }
+
+    private LocalDateTime commonTimeForm(LocalDateTime dateTime) {
+        if (dateTime.getMinute() % 60 < 30) {
+            return  LocalDateTime.of(dateTime.getYear(), dateTime.getMonth(),
```

Comment:
you implemented a half of my proposal :-)
save result into local field -> 
`int res = dateTime.getMinute() % 60;`
`return res > 30 ? dateTime.plusMinutes(60 - res) : dateTime.plusMinutes(30 - res)`

Reply:
Done. As it turned out we can not use the remainder of the division